From: Lucas Kanashiro <kanashiro@debian.org>
Date: Mon, 13 Apr 2020 14:40:16 -0300
Subject: Fix FTBFS on x32: misdetected as i386 or amd64

Author: Thorsten Glaser <t.glaser@tarent.de>
Bug: #954293

Updated by John Paul Adrian Glaubitz <glaubitz@physik.fu-berlin.de> on
2022-06-02, s/rb_cv_coroutine/coroutine_type/.
---
 configure.ac | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.ac b/configure.ac
index 5165735..ddbfe0f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2657,6 +2657,9 @@ AS_CASE([$coroutine_type], [yes|''], [
         [powerpc64-darwin*|ppc64-darwin*], [
             coroutine_type=ppc64
         ],
+        [x86_64-linux-gnux32], [
+            coroutine_type=ucontext
+        ],
         [x*64-linux*], [
             AS_CASE(["$ac_cv_sizeof_voidp"],
                 [8], [ coroutine_type=amd64 ],
